pgmpy/continuous/discretize/base.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import division
+
+from six import with_metaclass
+from abc import ABCMeta, abstractmethod
+
+import numpy as np
+
+from pgmpy.factors import ContinuousFactor
+
+
+class BaseDiscretizer(with_metaclass(ABCMeta)):
+    """
+    Base class for the discretizer classes in pgmpy. The discretizer
+    classes are used to discretize a continuous random variable
+    distribution into discrete probability masses.
+
+    Parameters
+    ----------
+    factor: ContinuousFactor
+        the continuous factor representing the distribution
+        to be discretized.
+    low, high: float
+        the range over which the function will be discretized.
+    cardinality: int
+        the number of states required in the discretized output.
+
+    """
+    def __init__(self, factor, low, high, cardinality):
+        if not isinstance(factor, ContinuousFactor):
+            raise ValueError("{factor} is not a valid ContinuousFactor object."
+                             .format(factor=factor))
+
+        self.factor = factor
+        self.low = low
+        self.high = high
+        self.cardinality = cardinality
+
+    @abstractmethod
+    def get_discrete_values(self):
+        """
+        This method implements the algorithm to discretize the given
+        continuous distribution.
+        It must be implemented by all the subclasses of BaseDiscretizer.
```

Comment:
@yashu-seth Try to add extra blank line after paragraphs. Makes things much more cleaner.